src/apps/external_executor/protobuf/index.proto

```diff
@@ -0,0 +1,74 @@
+syntax = "proto3";
+
+import "google/protobuf/empty.proto";
+
+package externalexecutor.protobuf;
+
+option optimize_for = LITE_RUNTIME;
+
+// Service exposed by a CCF node for Executors/Indexers to interact with the
+// Index stored on the CCF node. Communication with this service must be over
+// TLS, authorising as a client cert that has previously been accepted by the
+// ExecutorRegistration service.
+service Index
+{
+  // Installs index and registers with the given strategy name and table name.
+  // Streams Key-Value pairs of every transaction in the given table name.
+  // Stores indexed data in an unordered map datastructure or a prefix tree
+  // depending on the type of enum passed
+  rpc InstallAndSubscribe(IndexInstall) returns (stream IndexWork) {}
+
+  // Stores Indexer's data. Can only be invoked by the Indexer that installed
+  // the strategy
+  rpc StoreIndexedData(IndexKeyValue) returns (google.protobuf.Empty) {}
+
+  // Takes an input of Index Strategy name and a Key, returns values from the
+  // indexed data on the local CCF node This can be invoked by any registered
+  // executor
+  rpc GetIndexedData(IndexKey) returns (IndexKeyValue) {}
```

Comment:
I think there should be a way to unsubscribe the index where stream connection can be gracefully terminated. I will add that endpoint, but thoughts are welcome.